### ISSUE TITLE
Finishing `map` causes deadlock exception

### ISSUE THREAD

**Issue by akerbos** [grosser/parallel#88]
I execute some jobs with `Parallel.map`. After the tasks are completed (determined by some `puts`), that is immediately before or after the block passed to `map` ends, the program hangs for a second or two and then throws

    /celluloid-0.15.2/lib/celluloid/mailbox.rb:100:in `lock': deadlock detected (fatal)
    ...
    from /.../celluloid-0.15.2/lib/celluloid.rb:131:in `block in register_shutdown'
    from /.../parallel-0.9.1/lib/parallel.rb:291:in `fork'
    ...

I have `celluloid (0.15.2)` and `parallel (0.9.1)` installed. My system is `Linux 3.2.0-55-generic x86_64 GNU/Linux` with `ruby 1.9.3p0`.

Sadly, I can not reproduce the error with a small example. I have commented out all parts of my code that I thought may be at fault (including `puts`, the actual taks, etc) but the problem persists.

What may be wrong?

**Comment by akerbos** [comment]
Nowadays I get `Couldn't cleanly terminate all actors in 10 seconds!` (once per actor).

**Comment by grosser** [comment]
an example repo to reproduce this in might get us closer to a fix :)

**Comment by akerbos** [comment]
@grosser I'll try to rig together an MWE. You can check out [ltx2any](https://github.com/akerbos/ltx2any) if that's any help; running `ltx2any -d tikzext_test.tex` (you'll need a LaTeX installation for that, though). I get the errors almost every time. (Ubuntu 14.04 3.13.0-45-generic x84_64, ruby 1.9.3p484)

**Comment by akerbos** [comment]
@grosser That was easier than I feared: [MWE here](https://gist.github.com/akerbos/70e294bfb7aea943dbea). Fails every time for me.

**Comment by grosser** [comment]
The listen thing is not parallel fault, just listen/celluloid not working with forks

```Ruby
require 'listen'
Listen.to('.'){|*args|}.start
fork {}
# ... wait 10 seconds ...
# E, [2015-05-06T19:52:00.855864 #82819] ERROR -- : Couldn't cleanly terminate all actors in 10 seconds!
```

**Comment by akerbos** [comment]
@grosser Yes, it has been confirmed to be a bug in celluloid ([1](https://github.com/guard/listen/issues/169), [2](https://github.com/celluloid/celluloid/issues/612), [3](https://github.com/celluloid/celluloid/pull/523)). It has been patched; will be released with 0.16.1, apparently.

Thanks!
